projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -376,6 +378,28 @@ def pre_ingest_process(self,
             containing metadata produced by the pre-ingest plugin,
             and possibly used by other pre-ingest, ingest, or post-ingest
             plugins.
+            NOTE: It is possible to do modifications to dynamic parameters like
+            target or destination_table. However, when such modifications are
+            done, they need to be added to an `updated_dynamic_params`
+            dictionary. For example:
+                .. code-block:: python
+                    def pre_ingest_process(self,
+                        payload: List[dict],
+                        destination_table: Optional[str],
+                        target: Optional[str],
+                        collection_id: Optional[str],
+                        metadata: Optional[IngestionMetadata],
+                    ) -> Tuple[List[Dict], Optional[IngestionMetadata]]:
+                        if metadata:
+                            metadata["updated_dynamic_params"] = {
```

Comment:
I am not sure how much we want to change IngestionMetadata. But instead of string , they normal attributes we may have intellisense and such 
It is always better to design interfaces that prevent errors from happening.
 
Have you considered returning them in pre_ingest ? 

return payload, metadata, destination_table, target 


Or perhaps having class PreIngestResult { payload, metadata, destination, target } 

??

Reply:
I agree that changing IngestionMetadata is not optimal, and at first tried to extend the API with returning a dictionary containing all the updated params. However, due to the possibility to have multiple pre-processors, we would need to also include the data structure/type to the call arguments list. As I was making the changes, realised that this would be quite a breaking change, and it would probably break stuff used by custom pre-ingest plugins used in production environments. The also would make the API heavier than it already is. That's why I went for the most minimal change possible.

That being said, I agree that having a constant is much better choice than expecting users to remember how to type the key correctly. Code is updated for this.

Reply:
What do you mean by "make the API heavier" ?  Where is the term coming from? 

That being said the reasoning sounds good to me. Let's make target and destination constant (or enum) - they are pre-defined set of values that must be set correctly - candidate for constants.

Reply:
I meant it as in "developers need to account for an increasing number of call arguments, etc.". It is not an official term or anything like that.